src/commands/deployments/disconnectRepo.ts

```diff
@@ -3,13 +3,20 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DeploymentsTreeItem } from "vscode-azureappservice";
+import { DeploymentsTreeItem, disconnectRepo as disconnectRepository } from "vscode-azureappservice";
 import { IActionContext } from "vscode-azureextensionui";
+import { WebAppTreeItem } from '../../explorer/WebAppTreeItem';
 import { ext } from "../../extensionVariables";
+import { localize } from '../../localize';
 
 export async function disconnectRepo(context: IActionContext, node?: DeploymentsTreeItem): Promise<void> {
     if (!node) {
         node = <DeploymentsTreeItem>await ext.tree.showTreeItemPicker(DeploymentsTreeItem.contextValueConnected, context);
     }
-    await node.disconnectRepo(context);
+
+    if (node.parent instanceof WebAppTreeItem) {
```

Comment:
Same here, `instanceof SiteTreeItem`.  Otherwise you get the Action not supported error when trying to disconnect the repo from a deployment slot.